From 32e802f3085b391fb10d29cee0dd1f04e21570a7 Mon Sep 17 00:00:00 2001
From: Nidhi Jaju <nidhijaju@chromium.org>
Date: Thu, 6 Jun 2024 05:03:56 +0000
Subject: Backport uint32_t support for QUICHE_PROTOCOL_FLAG

Original message:
Roll src/net/third_party/quiche/src/ 656da99db..2b130243c (3 commits)

https://quiche.googlesource.com/quiche.git/+log/656da99db392..2b130243c5a1

$ git log 656da99db..2b130243c --date=short --no-merges --format='%ad %ae %s'
2024-06-05 quiche-dev Introduce new QUICHE_PROTOCOL_FLAG of type uint32_t
2024-06-04 danzh Limit the number of port migrations a QUIC client will attempt upon path degrading signals. This is to prevent the client from migrating port too frequently, which may cause port exhaustion.
2024-06-04 quiche-dev LSC: Move expensive variables on their last use to avoid copies.

Created with:
  roll-dep src/net/third_party/quiche/src src/third_party/quic_trace/src

Added support for uint32_t Quic Flags

Change-Id: Ia98845696c7b2fec673fc7f2eabbb17d66e85ae4
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5599463
Reviewed-by: Adam Rice <ricea@chromium.org>
Commit-Queue: Nidhi Jaju <nidhijaju@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1311075}
---
 net/quic/set_quic_flag.cc | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/net/quic/set_quic_flag.cc b/net/quic/set_quic_flag.cc
index 5ab2e7df1e263..e7990fe27c45d 100644
--- a/net/quic/set_quic_flag.cc
+++ b/net/quic/set_quic_flag.cc
@@ -32,10 +32,18 @@ void SetQuicFlagByName_float(float* flag, const std::string& value) {
   }
 }
 
+void SetQuicFlagByName_uint32_t(uint32_t* flag, const std::string& value) {
+  uint32_t val;
+  if (base::StringToUint(value, &val)) {
+    *flag = val;
+  }
+}
+
 void SetQuicFlagByName_uint64_t(uint64_t* flag, const std::string& value) {
   uint64_t val;
-  if (base::StringToUint64(value, &val) && val >= 0)
+  if (base::StringToUint64(value, &val)) {
     *flag = val;
+  }
 }
 
 void SetQuicFlagByName_int32_t(int32_t* flag, const std::string& value) {
-- 
2.43.0

